src/GraphQLParser/ParserContext.Parse.cs

```diff
@@ -250,7 +249,7 @@ private GraphQLDirectiveDefinition ParseDirectiveDefinition()
             ExpectKeyword("on");
             var locations = ParseDirectiveLocations();
 
-            return _ignoreOptions == IgnoreOptions.IgnoreCommentsAndLocations
+            return _ignoreOptions.HasFlag(IgnoreOptions.Locations)
```

Comment:
All of these conditions are wrong.  If only 'Locations' is specified to be skipped, 'Comments' are also skipped because the non-"full" version is used which does not have a field to store the comments node.

```suggestion
            return _ignoreOptions == IgnoreOptions.All
```